### DISCUSSION TITLE
How to use aws-iot-device-client to read/write to the Classic Shadow

### DISCUSSION THREAD

**Discussion by niccolo-caban** [aws/aws-iot-device-sdk-python#345] [Q&A]
Hi there! 
Does anyone know if its possible to read/write to the Classic Shadow (Default one) using aws-iot-device-client? 
I've been able to connect to other named shadows, but for some reason I can't use the Classic Shadow. 

I tried setting the **shadow-name** parameter to "Classic Shadow", but that didn't seem to work. I also left it blank, but then the client would error and say that I needed to provide a value. 

![Classic_Shadow](https://github.com/user-attachments/assets/e8e3e365-6913-4743-8537-49a931267508)

**Comment by jmklix** [comment]
We have a sample of using a classic shadow in the v1 to v2 [migration guide](https://github.com/aws/aws-iot-device-sdk-python-v2/blob/main/documents/MIGRATION_GUIDE.md#client-for-aws-iot-device-shadow):
```python
# Update Shadow
def customShadowCallback_Update(payload, responseStatus, token):
    return

JSONPayload = '{"state":{"desired":{"property":' + str(3) + '}}}'
deviceShadowHandler.shadowUpdate(
        JSONPayload,
        customShadowCallback_Update,
        5)
```

**Reply by niccolo-caban** [reply]
Gotcha, but can we use the classic shadow with the aws-iot-device-client?
We haven't been able to cross-compile **aws-iot-device-sdk-python-v2**, so we can only use aws-iot-device-client right now

**Reply by niccolo-caban** [reply]
We've been able to write to the classic shadow using the v1 Python SDK, but we'd like to do it using aws-iot-device-client instead

**Reply by jmklix** [reply]
Does @bretambrose [answer](https://github.com/aws/aws-iot-device-sdk-python/discussions/345#discussioncomment-10963659) work for you?

**Reply by niccolo-caban** [reply]
Unfortunately we lack the ability to compile the source code right now for our embedded linux platform, so we're limited to just using the aws-iot-device-client binary.

**Comment by bretambrose** [comment]
The device client is in C++ and uses the v2 C++ SDK.  So you should be able to put something together using https://github.com/aws/aws-iot-device-sdk-cpp-v2/tree/main/samples/shadow/shadow_sync
